_episodes/02-concepts.md

```diff
@@ -175,21 +252,59 @@ Lesson 5 covers how to use continuous integration services like Travis
 and CircleCI, but also how container technologies can be used to run 
 your own integration testing.
 
+> ## Exercise:
+>
+>  How does setting up continuous integration testing help with research? 
+>
+> > ## Solution
+> >
+> > During the lifetime of a project, the data and software may change. Setting 
+> > up testing environments allow ensuring that changes in results can be 
+> > attributed to changes in data and software. Results that remain the same across
+> > different software and data sets are more generalizable than results that 
```

Comment:
I somehow hesitate to explain CI taking changing data sets as an example... Since I see it couple times, you're doing this on purpose, but there are many "data set changes" that should lead to completely different outcome, not to generalization, so it's confusing IMO.